spill/board.c

```diff
@@ -17,94 +17,98 @@
 
 #include "snake.h"
 #include "board.h"
+#include "constants.h"
 #include <stddef.h>
 #include <stdlib.h>
 #include <string.h>
 #include <stdio.h>
+#include <stdint.h>
 
-int board[BOARD_HEIGHT][BOARD_WIDTH];
-
+long int board[BOARD_HEIGHT][BOARD_WIDTH];
+int count_food;
 
 void init_board(void) {
 	// Initialize the board with only 0s (clear blocks)
 	for (int i = 0; i < BOARD_HEIGHT; i++) {
 		for (int j = 0; j < BOARD_WIDTH; j++) {
-			board[i][j] = 0;
+			board[i][j] = BLOCK_BLANK;
 		}
 	}
 }
 
-int get_square_value(int x, int y) {
+long int get_square_value(int x, int y) {
     return board[y][x];
 }
 
-void set_square_value(int x, int y, int value) {
+void set_square_value(int x, int y, long int value) {
     board[y][x] = value;
 }
 
 // To be used under initialization, put a whole snake on the board
 void add_snake_to_board(struct Body *snake_head){
     struct Body *snake_part = snake_head;
     do {
-        set_square_value(snake_part->x, snake_part->y, BLOCK_SNAKE);
+        long int adr = (long long) snake_part;
+        set_square_value(snake_part->x, snake_part->y, adr);
+        struct Body* test = (struct Body*) get_square_value(snake_part->x, snake_part->y);
         snake_part = snake_part->next;
     } while (snake_part->next != NULL);
+    long int adr = (long int) snake_part;
+    set_square_value(snake_part->x, snake_part->y, adr);
```

Comment:
This is awfully hacky. storing pointers to objects as integers and putting them into the table is just not going to work. I think to do this properly you should make a new struct named BoardObject or BoardEntity or BoardItem or something which can store a pointer to a snake or information about what else is being put in the tile .

```
enum ObjectType (Blank, Food, Snake)

struct BoardObject {
    ObjectType type,
    Body* snake_adr
}
```

This would require a lot of reformating of the rest of the code, but I think it would be a lot simpler to work with.